python/semantic_kernel/connectors/memory/astradb/astradb_memory_store.py

```diff
@@ -0,0 +1,319 @@
+
+        if dimension_num > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {dimension_num} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+
+        result = await self._client.create_collection(
+            collection_name, dimension_num, distance_type
+        )
+        if result is True:
+            logger.info(f"Collection {collection_name} created.")
+
+    async def delete_collection_async(self, collection_name: str) -> None:
+        """Deletes a collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to delete.
+
+        Returns:
+            None
+        """
+        result = await self._client.delete_collection(collection_name)
+        logger.log(
+            logging.INFO if result is True else logging.WARNING,
+            f"Collection {collection_name} {'deleted.' if result is True else 'does not exist.'}",
+        )
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """Checks if a collection exists.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to check.
+
+        Returns:
+            bool -- True if the collection exists; otherwise, False.
+        """
+        return await self._client.find_collection(collection_name)
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """Upserts a memory record into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            record {MemoryRecord} -- The memory record to upsert.
+
+        Returns:
+            str -- The unique identifier for the memory record.
+        """
+        filter = {"_id": record._id}
+        update = {"$set": build_payload(record)}
+        status = await self._client.update_document(
+            collection_name, filter, update, True
+        )
+
+        return status["upsertedId"] if "upsertedId" in status else record._id
+
+    async def upsert_batch_async(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        """Upserts a batch of memory records into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            records {MemoryRecord} -- The memory records to upsert.
+
+        Returns:
+            List[str] -- The unique identifiers for the memory record.
+        """
+        return await asyncio.gather(
```

Comment:
why does this method not user update_documents? seems like there is support for native batching on the API side which should be much faster?

Reply:
There's an astra API for batching. But it can't update by the list of id.
`curl -s --location \
--request POST ${ASTRA_DB_API_ENDPOINT}/api/json/v1/${ASTRA_DB_KEYSPACE}/${COLLECTION} \
--header "Token: ${ASTRA_DB_APPLICATION_TOKEN}" \
--header "Content-Type: application/json" \
--header "Accept: application/json" \
--data '{
  "updateMany": {
    "filter": {"status" : "active" },
    "update" : {"$set" : { "status" : "inactive"}}
  }
}' | json_pp`

As you can see, can't pass multiple ids to filter option.

Reply:
@eavanvalkenburg

Reply:
done